### PR TITLE
Generate Terraform config from volumeset templates

### PR DESCRIPTION
**Pull request by zzaakiirr**
### What does this PR do?

This PR adds support for converting templates with type: volumeset to terraform config format

### Terraform docs
https://registry.terraform.io/providers/controlplane-com/cpln/latest/docs/resources/volume_set

### Examples

```
kind: volumeset
name: postgres-poc-vs
description: postgres-poc-vs
spec:
  autoscaling:
    maxCapacity: 1000
    minFreePercentage: 1
    scalingFactor: 1.1
  fileSystemType: ext4
  initialCapacity: 10
  performanceClass: general-purpose-ssd
  snapshots:
    createFinalSnapshot: true
    retentionDuration: 7d
```

Transforms to:
```
resource "cpln_volume_set" "postgres-poc-vs" {
  gvc = cpln_gvc.app-name.name
  name = "postgres-poc-vs"
  description = "postgres-poc-vs"
  initial_capacity = 10
  performance_class = "general-purpose-ssd"
  file_system_type = "ext4"
  snapshots {
    create_final_snapshot = true
    retention_duration = "7d"
  }
  autoscaling {
    max_capacity = 1000
    min_free_percentage = 1
    scaling_factor = 1.1
  }
}
```

## Release Notes

- **New Features**
	- Introduced the `VolumeSet` class for managing volume configurations in Terraform.
	- Added a new `pluralize` method for string manipulation.

- **Bug Fixes**
	- Improved error handling for unsupported template kinds and configuration generation.

- **Documentation**
	- Expanded RSpec tests for `VolumeSet` and `String` classes, ensuring comprehensive coverage and validation.

- **Refactor**
	- Enhanced error management and type handling in the `Generate` and `Generator` classes.
	- Updated template handling and configuration management within the `Generator` class.

### CODE REVIEW

**Comment by zzaakiirr** [comment]
@coderabbitai review

File: spec/core/terraform_config/volume_set_spec.rb [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        expect(generated).to eq(
+          <<~EXPECTED
+            resource "cpln_volume_set" "#{options.fetch(:name)}" {
+              gvc = "#{options.fetch(:gvc)}"
+              name = "#{options.fetch(:name)}"
+              description = "#{options.fetch(:description)}"
+              tags = {
+                env = "test"
+                project = "example"
+              }
```
[the remark this answers is not in the document]
Comment (zzaakiirr) [reply spec/core/terraform_config/volume_set_spec.rb:34]: Current version is better because suggested version is not handling well indendations

**Comment by zzaakiirr** [comment]
@coderabbitai review

File: lib/core/terraform_config/volume_set.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      return if PERFORMANCE_CLASSES.include?(performance_class.to_s)
+
+      raise ArgumentError,
+            "Invalid performance class: #{performance_class}. Choose from #{PERFORMANCE_CLASSES.join(', ')}"
+    end
+
+    def validate_file_system_type!
+      return if FILE_SYSTEM_TYPES.include?(file_system_type)
+
+      raise ArgumentError, "Invalid file system type: #{file_system_type}. Choose from #{FILE_SYSTEM_TYPES.join(', ')}"
```
[the remark this answers is not in the document]
Comment (zzaakiirr) [reply lib/core/terraform_config/volume_set.rb]: Done

**Review by rafaelgomesxyz — CHANGES_REQUESTED** [review]

File: spec/patches/string_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    context "when word ends with 'y'" do
+      it "changes 'y' to 'ies'" do
+        expect("city".pluralize).to eq("cities")
+        expect("baby".pluralize).to eq("babies")
+      end
+    end
+
+    context "when word doesn't end with 'y', 's', 'x', 'z', 'ch', or 'sh'" do
+      it "adds 's' to the end if it doesn't already end with 's'" do
```
[the remark this answers is not in the document]
Comment (zzaakiirr) [reply spec/patches/string_spec.rb]: Done

File: lib/patches/string.rb [hunk 4]
```diff
@@ -22,5 +22,19 @@ def unindent
...
+    return self if empty?
+
+    if end_with?("ies")
+      self
+    elsif end_with?("s", "x", "z", "ch", "sh")
+      end_with?("es") ? self : "#{self}es"
+    elsif end_with?("y")
+      "#{self[...-1]}ies"
+    else
+      end_with?("s") ? self : "#{self}s"
```
[the remark this answers is not in the document]
Comment (zzaakiirr) [reply lib/patches/string.rb]: Simplified method and left comment for this method

File: lib/core/terraform_config/generator.rb [hunk 5]
```diff
@@ -2,69 +2,85 @@
 
 module TerraformConfig
   class Generator
+    SUPPORTED_TEMPLATE_KINDS = %w[gvc secret identity policy volumeset].freeze
+
+    InvalidTemplateError = Class.new(ArgumentError)
```
[the remark this answers is not in the document]
Comment (zzaakiirr) [reply lib/core/terraform_config/generator.rb]: Done

**Review by rafaelgomesxyz — APPROVED** [review]
LGTM

Waiting on review from @borela.

**Review by borela — APPROVED** [review]

**Pull request merged**